CMakeLists.txt

```diff
@@ -91,6 +74,56 @@ if(ENABLE_MKLCPU_BACKEND
   list(APPEND DOMAINS_LIST "rng")
 endif()
 
+# Define required CXX compilers before project
+if(CMAKE_CXX_COMPILER)
+  if(WIN32)
+    string(REPLACE "\\" "/" CMAKE_CXX_COMPILER ${CMAKE_CXX_COMPILER})
+  endif()
+else()
+  if(ENABLE_CUBLAS_BACKEND OR ENABLE_CURAND_BACKEND)
+    set(CMAKE_CXX_COMPILER "clang++")
+  elseif(ENABLE_MKLGPU_BACKEND)
+    set(CMAKE_CXX_COMPILER "dpcpp")
+  else()
+    find_program(DPCPP_PATH dpcpp)
+    if(DPCPP_PATH)
+      set(CMAKE_CXX_COMPILER "dpcpp")
+    else()
+      if(WIN32)
+        message(STATUS "dpcpp was not found in PATH, using clang-cl instead")
+        set(CMAKE_CXX_COMPILER "clang-cl")
+      else()
+        message(STATUS "dpcpp was not found in PATH, using clang++ instead")
+        set(CMAKE_CXX_COMPILER "clang++")
+      endif()
+    endif()
+  endif()
+endif()
+
+# Define required C compilers for Windows before project
+if(WIN32)
+  set(CMAKE_C_COMPILER "clang-cl")
+endif()
+
+project(oneMKL VERSION 0.1.0 LANGUAGES CXX)
+
+# Override default CXX compile/link lines for Windows after project
+if(WIN32)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-function -w")
+  foreach (flag_var
+           CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
+           CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
+    string(REPLACE "/MD" "" ${flag_var} "${${flag_var}}")
+  endforeach()
+  set(CMAKE_CXX_COMPILE_OBJECT "<CMAKE_CXX_COMPILER> -fsycl /nologo <DEFINES> <INCLUDES> /EHsc <FLAGS> /Fo<OBJECT> -c <SOURCE>")
+  set(CMAKE_CXX_CREATE_STATIC_LIBRARY "lib /nologo <OBJECTS> /out:<TARGET>")
+  set(CMAKE_CXX_LINK_EXECUTABLE "<CMAKE_CXX_COMPILER> -fsycl /nologo <OBJECTS> -o <TARGET> <LINK_LIBRARIES>")
```

Comment:
In the past, overriding of `CMAKE_CXX_*` options was required, because Intel oneAPI DPC++ Compiler did not have official support in CMake, but things have changed a little since CMake 3.20.1. Which CMake version did you use for your local testing?

Reply:
I used CMake 3.16.3 on Linux and 3.15.7 on Windows. I think it would be better not to rely on this feature because the minimum version we require is 3.13.

Reply:
My point is that we saw some warnings and CMake policy errors when using 3.20.1 when it was released, so it is better to check with CMake 3.20.1 also, that overriding default CMake behavior is fine.

Reply:
It works with CMake 3.20.1. I was wrong about the version on Windows. `setvars.bat` changes the version to 3.20.21032501-MSVC_2.

Reply:
okay thanks!